infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutTrackNumberService.kt

```diff
@@ -198,34 +198,32 @@ class LayoutTrackNumberService(
         polygonBufferSize: Double,
         startKmNumber: KmNumber?,
         endKmNumber: KmNumber?,
-    ): List<GeometryPlanHeader> {
-        val referenceLine = referenceLineService.getByTrackNumber(layoutContext, trackNumberId)
-        if (referenceLine == null) {
-            return emptyList()
-        }
-
-        val alignmentVersion = requireNotNull(referenceLine.alignmentVersion)
-        val geocodingContext = requireNotNull(geocodingService.getGeocodingContext(layoutContext, trackNumberId))
-
-        if (!cropIsWithinReferenceLine(startKmNumber, endKmNumber, geocodingContext)) {
-            return emptyList()
-        } else if (startKmNumber == null && endKmNumber == null) {
-            return alignmentService.getOverlappingPlanHeaders(
-                alignment = layoutAlignmentDao.fetch(alignmentVersion),
-                polygonBufferSize = polygonBufferSize,
-                cropStartM = null,
-                cropEndM = null,
-            )
-        } else {
-            val (cropStartM, cropEndM) = getCropMValues(geocodingContext, startKmNumber, endKmNumber)
-            return alignmentService.getOverlappingPlanHeaders(
-                alignment = layoutAlignmentDao.fetch(alignmentVersion),
-                polygonBufferSize = polygonBufferSize,
-                cropStartM = cropStartM,
-                cropEndM = cropEndM,
-            )
-        }
-    }
+    ): List<GeometryPlanHeader> =
+        referenceLineService.getByTrackNumber(layoutContext, trackNumberId)?.let { referenceLine ->
+            val alignmentVersion =
+                requireNotNull(referenceLine.alignmentVersion) { "Reference line must have an alignment" }
+
+            geocodingService.getGeocodingContext(layoutContext, trackNumberId)?.let { geocodingContext ->
+                if (!cropIsWithinReferenceLine(startKmNumber, endKmNumber, geocodingContext)) {
+                    emptyList()
+                } else if (startKmNumber == null && endKmNumber == null) {
```

Comment:
Tämä ja alempi haara voisi vain yhdistää. Koska tämä haara vaatii että molemmat on null, tuo alempi ei voi olettaa kummankaan olevan non-null -> getCropMValues ottaa molemmat argsit nullablena. Jos ne on molemmat null, sieltä palautuu `(null to null)` -> päätyy kutsumaan getOverlappingPlanHeaders-funkkaria tismalleen samoilla argumenteilla kuin tämäkin.